### DISCUSSION TITLE
How can I migrates my game ports saves to v2.0+ ?

### DISCUSSION THREAD

**Discussion by Softhook** [eduardofilo/RG350_adam_image#286] [Ports]
How can I migrates my native game port saves to v2.0+ ?

I have a save for Cave story and Diabolo that I would be very sad to loose.

Also how does V2.0+ deal with saves for ports in general?

Thanks for your help ... :)

**Comment by eduardofilo** [comment]
The Cave Story ones are in the `/media/data/local/home/.nxengine` directory. The savestates correspond with the `profileN.dat` files.

![screenshot001](https://user-images.githubusercontent.com/2923728/182721041-0df89aa7-53a9-4aae-94b4-31a3455eadda.png)

Diabolo I have never used, but I suppose that like most OpenDingux applications, it will keep its files in a hidden directory in home named `.diabolo` or something similar.

**Reply by Softhook** [reply]
Thanks very much for that.

So just to be clear - v2.0+ any update will always require manual moving of port saves?
Are there any plans to move them to the external SD as well?

thanks :)

**Reply by eduardofilo** [reply]
In general, yes. The problem with the ports is that each one is programmed in a different way and it is not possible to generalize the way to divert the savestates to the external card. The only way would be to divert the entire home directory, which could have many problems as FAT systems do not support Linux symbolic links. Another way would be to make the image upgradable in another way without doing a full flash, but it is not in my plans to study that. Sorry.

**Reply by Softhook** [reply]
thanks very much for the clarity!
